contracts/flashloan/Instapool/main.sol

```diff
@@ -101,28 +106,15 @@ contract DydxFlashloaner is Helper, ICallee, DydxFlashloanBase {
         }
     }
 
-    function routeDydx(address token, uint256 amount, bytes memory data) internal {
+    function routeDydx(address token, uint256 amount, uint256 route, bytes memory data) internal {
         uint256 _amount = amount;
 
-        uint256 route = 0;
-        address _token;
-        if (token == daiAddr) {
-            uint256 dydxDaiAmt = daiContract.balanceOf(soloAddr);
-            if (amount > dydxDaiAmt) {
-                uint256 dydxWEthAmt = wethContract.balanceOf(soloAddr);
-                route = 1;
-                _amount = sub(dydxWEthAmt, 10000);
-                _token = wethAddr;
-            } else {
-                _token = token;
-            }
-        } else if (token != ethAddr && token != usdcAddr && token != wethAddr) {
+        address _token = token == ethAddr ? wethAddr : token;
+        if (route != 0) {
             uint256 dydxWEthAmt = wethContract.balanceOf(soloAddr);
             route = 2;
```

Comment:
@KaymasJain Do we need this?

Reply:
We are already passing the route, right?

Reply:
That's why I removed it! We can skip it.